### PR TITLE
Loot crate for health

### LINKED ISSUE #745: [FEATURE] Crate - Health
## 🚀 Feature
Implement all needed for Health Crate

- [ ] Spawn
- [ ] Grab on walk 
- [ ] Increase life

The code should be open so then it's easy to add other types of crates

### PR DESCRIPTION
**Pull request by AminArria**
Closes #745 #450 #406 #279

- [x] Spawn loot for health crate in random position healing a random amount between 25 and 75
- [x] Player pickups loot while moving with joystick
- [x] Client shows/remove loot
- [x] GameSetting to tune loot spawn rate

Pending for front
- [x] Proper model for the health loot. The `BattleManager` prefab has the script `Loot` which has a field to select the model for the health loot
- [x] Create heal particle effect
- [x] Create pooler fot the loot boxes

### CODE REVIEW

**Review by ncontinanza — CHANGES_REQUESTED** [review]
Nice job! Just a few comments

File: client/Assets/Scripts/Loot.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    public void RemoveLoots(List<LootPackage> updatedLoots)
+    {
+        var idsToRemove = this.loots.Keys.Except(updatedLoots.Select(loot => loot.Id)).ToList();
+        idsToRemove.ForEach(RemoveLoot);
+    }
+
+    private GameObject getLootObject(LootType lootType)
```
[the remark this answers is not in the document]
Comment (BertovDev) [reply client/Assets/Scripts/Loot.cs:52]: You are right!! thanks

File: client/Assets/Scripts/SocketConnectionManager.cs [hunk 2]
```diff
@@ -156,6 +157,8 @@ private void OnWebSocketMessage(byte[] data)
                     eventsBuffer.AddEvent(game_event);
                     this.gameProjectiles = game_event.Projectiles.ToList();
                     alivePlayers = game_event.Players.ToList().FindAll(el => el.Health > 0);
+                    // updateLoots(game_event.Loots.ToList());
```
[the remark this answers is not in the document]
Comment (BertovDev) [reply client/Assets/Scripts/SocketConnectionManager.cs:160]: Yess

File: client/Assets/Scripts/OverlayEffect.cs [hunk 3]
```diff
@@ -6,8 +6,9 @@
 
 public class OverlayEffect : MonoBehaviour
 {
-    public Shader shader;
-
+    public Shader DamageShader;
+    public Shader Healshader;
```
Comment (mgrunwaldt) [inline client/Assets/Scripts/OverlayEffect.cs:10]:
```suggestion
    public Shader HealShader;
```

File: client/Assets/Scripts/PlayerFeedbacks.cs [hunk 4]
```diff
@@ -40,7 +41,20 @@ public void ChangePlayerTextureOnDamage(GameObject player, float auxHealth, floa
 
         if (auxHealth != playerHealth)
```
Comment (mgrunwaldt) [inline client/Assets/Scripts/PlayerFeedbacks.cs:42]: not sure I understand what auxHealth and playerHealth represent, can you explain in the comment?

**Review by mgrunwaldt — CHANGES_REQUESTED** [review]
Good job. I left a few changes + a question. Besides that, its not looteable, its lootable, we should change it throughout the code,  (classes, uses in code, assets)

File: client/Assets/Scripts/PlayerFeedbacks.cs (hunk 4 above)
Comment (BertovDev) [reply client/Assets/Scripts/PlayerFeedbacks.cs:42]: `auxHealth` is the client health and `playerHealth` is the server health. We are checking if their are different so we can display the damage feedback and then execute `healthComponent.SetHealth(playerUpdate.Health);` to match them.

File: client/Assets/Scripts/Lootable.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "New Looteable", menuName = "CoM Looteable")]
```
Comment (ncontinanza) [inline client/Assets/Scripts/Lootable.cs:5]:
```suggestion
[CreateAssetMenu(fileName = "New Lootable", menuName = "CoM Lootable")]
```

File: client/Assets/Scripts/LootableList.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(menuName = "CoM Looteable List", fileName = "Looteable List")]
```
Comment (ncontinanza) [inline client/Assets/Scripts/LootableList.cs:5]:
```suggestion
[CreateAssetMenu(menuName = "CoM Lootable List", fileName = "Lootable List")]
```

**Review by ncontinanza — CHANGES_REQUESTED** [review]
Just a few more name changes and we're good to go.
Also look into renaming client/Assets/ScriptableObjects/Lootables/Looteable List.asset

**Review by ncontinanza — DISMISSED** [review]
LGTM! 🚀

**Comment by saugon** [comment]
Nice job! It's working as expected but there is something we need to change. @BertovDev

Loot.cs is currently managing the boxes, independent of the battle update logic (which is the one that handles updates between server and client). We should move the loot boxes management to Battle.cs, instead of having it in the Loot Update method.

Spoiler alert: there will be some improvements and changes after polishing some definitions around loot boxes 😁

**Review by saugon — DISMISSED** [review]
Good work! 🚀

**Review by ncontinanza — DISMISSED** [review]
Nice! 🚀

**Review by ncontinanza — APPROVED** [review]

**Review by saugon — APPROVED** [review]
Good work! 🚀

**Review by ncontinanza — APPROVED** [review]

**Pull request merged**